### PR TITLE
Added Volley stetho network

### PR DESCRIPTION
**Pull request by briangriffey**

### CODE REVIEW

**Comment by jasta** [comment]
I think there's a much easier way to do this by subclassing the two HttpStack implementations or maybe even easier by just creating a StethoWrappingHttpStack.

Also, prefer using Volley through Maven coordinates.  We don't currently bundle any jar files.

**Comment by briangriffey** [comment]
Extending HurlStack might be possible. That private static entityFromConnection might prove painful when trying to read the inputstream.

**Comment by jasta** [comment]
As I think about this more, I think wrapping HttpStack is more flexible and simpler.  This can be done by replacing the entity like:
```java
  public HttpResponse performRequest(...) {
    mNetworkEventReporter.requestWillBeSent(new VolleyInspectorRequest(request));
    response = mDelegate.performRequest(request);
    mNetworkEventReporter.receivedResponseHeaders(...);
    InputStream intercepted = mNetworkEventReporter.interpretResponseStream(...);
    response.setEntity(new InputStreamEntity(intercepted));
    return response;
  }
```
This should make it possible to do:
```java
new StethoInterceptingHttpStack(new HurlStack(...));
```

**Comment by jasta** [comment]
Just saw #43 which might be of interest to you as well.  This is a slightly different take on the solution if you use OkHttp.

**Comment by briangriffey** [comment]
Yea, I just think this is the only connection agnostic method to integrating with volley. Still allows you to use OkHttp or UrlConnection or HttpClient under the hood. This sits on the layer right above that.

**Comment by jasta** [comment]
@briangriffey agreed, I think the `HttpStack` wrapper works nicely for that reason too.

File: settings.gradle [hunk 1]
```diff
@@ -1,4 +1,5 @@
 include ':stetho'
```
Comment (mannyman1115) [inline settings.gradle]: Leave a line note

File: stetho-volley/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (mannyman1115) [inline stetho-volley/src/main/AndroidManifest.xml]: Leave a line note

**Comment by jasta** [comment]
Any update?  I'd really love to have good Volley support for Stetho...

**Comment by briangriffey** [comment]
Sorry, end of the quarter is happening. I'll circle back to this beginning of April.

**Comment by gfx** [comment]
👀

**Comment by milek** [comment]
Any update on that pull request?

**Comment by condemoreloer** [comment]
brian@briangriffey.com
Ledo un gouache.

**Comment by vaurdan** [comment]
Any news on this one? It is possible to use Stetho with Volley?

**Comment by jasta** [comment]
@vaurdan yes it's always been possible to use Stetho with Volley it just requires you jump through some hoops.  The easiest way is to use OkHttp with Volley and install this implementation into Volley: https://gist.github.com/bryanstern/4e8f1cb5a8e14c202750

This works great actually, it's just not as discoverable/convenient as we'd like.  It also doesn't work if you use Volley with HttpURLConnection :\

**Comment by vaurdan** [comment]
@jasta Oh! Thanks for the tip, I've now Stetho working with Volley using the OkHttp! :+1:

**Comment by YusukeIwaki** [comment]
With stetho-volley, NullPointerException occurs when the response code is 204.

Applying the patch below is required.
https://android.googlesource.com/platform/frameworks/volley/+/e8f5353b17aea6659dc3aa13223317b484d01483%5E%21/src/com/android/volley/toolbox/BasicNetwork.java

**Comment by shibenli** [comment]
Thanks to (http://www.littlerobots.nl/blog/stetho-for-android-debug-builds-only/ )[http://www.littlerobots.nl/blog/stetho-for-android-debug-builds-only/ ].I've now Stetho working with Volley using the OkHttp in my **debug** build, but using Volley alone  in **release**! :+1:

**Comment by jasta** [comment]
Ancient pull request and really isn't everyone just using OkHttp now? :)

**Pull request closed without merging**